src/app/user/(logged-in)/my-account/page.tsx

```diff
@@ -1,27 +1,101 @@
 "use server";
 
-import { BarkH1 } from "@/components/bark/bark-typography";
 import { getAuthenticatedUserActor } from "@/lib/auth";
 import { RoutePath } from "@/lib/route-path";
-import { UserPii } from "@/lib/data/db-models";
 import { redirect } from "next/navigation";
+import { Button, buttonVariants } from "@/components/ui/button";
+import { IMG_PATH } from "@/lib/image-path";
+import { getMyLatestCall } from "@/lib/user/actions/get-my-latest-call";
+import { getMyAccount } from "@/lib/user/actions/get-my-account";
+import { format, formatDistanceStrict } from "date-fns";
+import { BarkH1, BarkH4 } from "@/components/bark/bark-typography";
+import Link from "next/link";
+import Image from "next/image";
+
+import { capitalize } from "lodash";
 
 export default async function Page() {
   const actor = await getAuthenticatedUserActor();
   if (!actor) {
     redirect(RoutePath.USER_LOGIN_PAGE);
   }
-  const ownPii: UserPii | null = await actor.getOwnUserPii();
-  if (!ownPii) {
+  const account = await getMyAccount(actor);
+  if (!account) {
     redirect(RoutePath.USER_LOGIN_PAGE);
   }
+  const {
+    userId,
+    userCreationTime,
+    userResidency,
+    ownPii: { userName, userEmail, userPhoneNumber },
+  } = account;
+
+  let latestCall = (await getMyLatestCall(actor))?.callCreationTime;
+  const latestCallText = latestCall
+    ? formatDistanceStrict(latestCall, new Date(), { addSuffix: true })
+    : "N.A";
+
+  const userCreationTimeText = format(userCreationTime, "dd MMM yyyy");
+
   return (
-    <>
-      <BarkH1>My Account</BarkH1>
-      <div>
-        My Own PII
-        <pre>{JSON.stringify(ownPii, null, 2)}</pre>
+    <main className="flex flex-col gap-6">
```

Comment:
q - what does `main` mean?

Reply:
`main` functionally is the same as a `div` but adds semantic meaning to the structure of the document. It is useful for SEO and page-readers that use these tags to understand the flow of the page.

From what I read, each page should contain a `<main>` tag to denote the main content of the page. Other semantic tags include `<section>`, `<footer>` etc. - [Article](https://www.semrush.com/blog/semantic-html5-guide/?kw=&cmp=AA_SRCH_DSA_Blog_EN&label=dsa_pagefeed&Network=g&Device=c&utm_content=676513944951&kwid=dsa-2185834090056&cmpid=18361936995&agpid=162954371228&BU=Core&extid=105279941410&adpos=&gad_source=1&gclid=Cj0KCQjw5cOwBhCiARIsAJ5njuZS8hKDIdGKtTIX_Z1zvb1MqEd8m3L8egjOLns7H22WkXYOACeCugAaAhnGEALw_wcB)

Reply:
Thanks for the reference. TIL.